packages/editor/src/components/block-settings-menu/content-only-settings-menu.js

```diff
@@ -19,53 +20,81 @@ import { store as editorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
 function ContentOnlySettingsMenuItems( { clientId, onClose } ) {
-	const { entity, onNavigateToEntityRecord } = useSelect(
-		( select ) => {
-			const {
-				getBlockEditingMode,
-				getBlockParentsByBlockName,
-				getSettings,
-				getBlockAttributes,
-			} = select( blockEditorStore );
-			const contentOnly =
-				getBlockEditingMode( clientId ) === 'contentOnly';
-			if ( ! contentOnly ) {
-				return {};
-			}
-			const patternParent = getBlockParentsByBlockName(
-				clientId,
-				'core/block',
-				true
-			)[ 0 ];
+	const { entity, onNavigateToEntityRecord, canEditTemplateParts } =
+		useSelect(
+			( select ) => {
+				const {
+					getBlockEditingMode,
+					getBlockParentsByBlockName,
+					getSettings,
+					getBlockAttributes,
+				} = select( blockEditorStore );
+				const contentOnly =
+					getBlockEditingMode( clientId ) === 'contentOnly';
+				if ( ! contentOnly ) {
+					return {};
+				}
+				const patternParent = getBlockParentsByBlockName(
+					clientId,
+					'core/block',
+					true
+				)[ 0 ];
 
-			let record;
-			if ( patternParent ) {
-				record = select( coreStore ).getEntityRecord(
-					'postType',
-					'wp_block',
-					getBlockAttributes( patternParent ).ref
+				let record;
+				const _canEditTemplateParts = select( coreStore ).canUser(
+					'create',
+					'templates'
 				);
-			} else {
-				const { getCurrentPostType, getCurrentTemplateId } =
-					select( editorStore );
-				const currentPostType = getCurrentPostType();
-				const templateId = getCurrentTemplateId();
-				if ( currentPostType === 'page' && templateId ) {
+				if ( patternParent ) {
 					record = select( coreStore ).getEntityRecord(
 						'postType',
-						'wp_template',
-						templateId
+						'wp_block',
+						getBlockAttributes( patternParent ).ref
 					);
+				} else {
+					const { getCurrentTemplateId } = select( editorStore );
+					const templateId = getCurrentTemplateId();
+					if ( templateId && _canEditTemplateParts ) {
+						record = select( coreStore ).getEntityRecord(
+							'postType',
+							'wp_template',
+							templateId
+						);
+					}
 				}
-			}
-			return {
-				entity: record,
-				onNavigateToEntityRecord:
-					getSettings().onNavigateToEntityRecord,
-			};
-		},
-		[ clientId ]
-	);
+				return {
+					canEditTemplateParts: _canEditTemplateParts,
+					entity: record,
+					onNavigateToEntityRecord:
+						getSettings().onNavigateToEntityRecord,
+				};
+			},
+			[ clientId ]
+		);
+
+	if ( ! canEditTemplateParts ) {
+		return (
+			<>
+				<BlockSettingsMenuFirstItem>
+					<MenuItem
```

Comment:
So this was hacked together and needs review.

There are two items — MenuItem and Text — because there is a line separating the two that persists.

Ideally, the dropdown wouldn't be required at all and we'd replace the ellipsis menu with a lock icon.

Ran out of time to investigate that today.

Reply:
I don't think the lock icon is desired, as users can still edit the content of blocks like Post Title, even with low permissions. So this is on the right track, but the copy needs to be adjusted.

I think some text like "Only users with permissions to edit the template can move or delete this block" is more in the spirit of https://github.com/WordPress/gutenberg/pull/61127. Perhaps with an aria-disabled 'Edit Template' button.

@jameskoster may have some suggestions.

TLDR for you - the question is around what to show in list view here when the user doesn't have permissions to edit the template:
![Screenshot 2024-06-06 at 3 03 25 pm](https://github.com/WordPress/gutenberg/assets/677833/4da6e70d-0e32-40a2-b4f8-ef39052a9436)

Reply:
> I don't think the lock icon is desired, as users can still edit the content of blocks like Post Title, even with low permissions. So this is on the right track, but the copy needs to be adjusted.
> 
> I think some text like "Only users with permissions to edit the template can move or delete this block" is more in the spirit of https://github.com/WordPress/gutenberg/pull/61127.

I like this. I've updated the PR so that we're using the existing component, but disabling the button and adding the suggested text for now, until further design guidance 😄